src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.hpp

```diff
@@ -1552,6 +1576,7 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_AddSrpService(c
 template <class ImplClass>
 CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_RemoveSrpService(const char * aInstanceName, const char * aName)
 {
+    VerifyOrReturnError(mOTInst, CHIP_ERROR_INCORRECT_STATE);
```

Comment:
The SRP functions already check for `mSrpClient.mIsInitialized`, so no extra checks are needed, right?

Reply:
Keep the check in `ClearSrpHost()` and `ClearAllSrpHostAndServices()` which might be called before mIsInitialized is set to true and remove other.